lib/jxl/modular/transform/squeeze.cc

```diff
@@ -43,36 +185,105 @@ Status InvHSqueeze(Image &input, uint32_t c, uint32_t rc, ThreadPool *pool) {
     return true;
   }
 
+#if HWY_TARGET != HWY_SCALAR
```

Comment:
I'm mostly concerned about overflows in $weird situations in Squeeze tendency :)

Reply:
Fair enough. I'll add a test using high bit depth + ycocg + squeeze. I expect the breaking point to be a bit earlier with the int32_t simd code than with the scalar SmoothTendency, but I assume it will be past the 24-bit limit that we have as an implementation limit anyway for uint image types. But you're right that it's probably a good idea to check that with a test.

Reply:
I added the test in a separate PR: https://github.com/libjxl/libjxl/pull/1267

Once that one lands, I'll rebase this one so we can check that the test still succeeds.

Reply:
Rebased it, so we'll soon see if the test breaks or not.